### PR TITLE
Replace Travis with GitHub Actions

### LINKED ISSUE #50: Replace Travis with GitHub Actions
While there are are [many reasons to move away from Travis](https://twitter.com/asbjornu/status/1098873841273708544), the most important one is that Travis has stopped notifying GitHub (see travis-ci/travis-ci#10204). Instead of fixing the Travis checks, I think it's better to just replace it altogether with GitHub Actions.

The rationale to use GitHub Actions is that we're dependent on GitHub for hosting of the source code anyway and GitHub Actions is considered by many (myself included) to be simpler, more feature-rich and more enjoyable than just about every existing alternative.

### PR DESCRIPTION
**Pull request by asbjornu**
Replace Travis with GitHub Actions. Resolved #50.

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 2.1.401
+
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build
+      run: dotnet build --configuration Release --no-restore
```
Comment (tpluscode) [inline .github/workflows/dotnet.yml:25]: are those necessary? `dotnet test` should restore and build first, shouldn't it?

Comment (asbjornu) [reply .github/workflows/dotnet.yml:25]: I think separating it makes it easier to figure out where the problem is if something bombs.

**Comment by asbjornu** [comment]
> Looking good but why the sharp drop in coverage? 🤔

I'm dumbfounded by that too. 🤷‍♂️

**Comment by goofballLogic** [comment]
Why switch away from Travis?

**Comment by asbjornu** [comment]
@goofballLogic, [there are many reasons](https://twitter.com/asbjornu/status/1098873841273708544), but the most important one is that Travis has stopped notifying GitHub (see travis-ci/travis-ci#10204) so instead of fixing Travis, I think it's better to just replace it with GitHub Actions.

**Comment by goofballLogic** [comment]
That is very sad, but reading the background it appears to be the right call. I think we should have an issue for this to track the decision making process?

**Comment by asbjornu** [comment]
@goofballLogic, sure. I've created #50.

**Comment by asbjornu** [comment]
An approval of this PR would be highly appreciated, @tpluscode, @goofballLogic. 😅

**Comment by goofballLogic** [comment]
Don't we need to fix the test coverage?

**Comment by asbjornu** [comment]
>Don't we need to fix the test coverage?

As the test coverage reduction has nothing to do with this PR, there's nothing I can or will do about it in this PR. That's not to say we shouldn't do anything about it ever, it's just completely unrelated to this PR, which we need to get merged to even get code coverage reporting working again. :)

**Comment by goofballLogic** [comment]
ok i guess we just disable the check for now...

**Comment by asbjornu** [comment]
The code coverage check isn't required to pass, so there's no need to disable it. Thanks for the approval, @goofballLogic!

**Pull request merged**